File: app/src/main/java/com/example/goesplayer/Music.java
package com.example.goesplayer;

import android.net.Uri;

public class Music {

    private long idNumber;
    private String name;
    private String title;
    private String artist;
    private String album;
    private String gender;
    private String folder;
    private Uri uri;

    public Music(long id, String archiveName, String musicName, String artistName, String albumName, String gender, String folder, Uri uri) {
        this.idNumber = id;
        this.name = archiveName;
        this.title = musicName;
        this.artist = artistName;
        this.album = albumName;
        this.gender = gender;
        this.folder = folder;
        this.uri = uri;
    }

    public long getIdNumber(){
        return idNumber;
    }

    public String getName() {
        return name;
    }

    public String getTitle() {
        return title;
    }

    public String getArtist() {
        return artist;
    }

    public String getAlbum() {
        return album;
    }

    public String getGenre() {
        return gender;
    }

    public String getFolder() {
        return folder;
    }

    public Uri getUri() {
        return uri;
    }
}

File: app/src/main/java/com/goesplayer/data/model/Music.kt
package com.goesplayer.data.model

import android.net.Uri

data class Music(
    var id: Long,
    val fileName: String,
    val title: String,
    val artist: String,
    val album: String,
    val genre: String,
    val songUri: Uri,
    val albumArtUri: Uri?,
    val durationInSeconds: Long
) {
    val folder: String?
        get() {
            val path = songUri.path?.removeSuffix(fileName)
            return path?.substring(path.lastIndexOf("/") + 1)
        }
}